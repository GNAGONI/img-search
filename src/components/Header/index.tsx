import { styled } from '@mui/material/styles';

type Props = {
  children: React.ReactNode;
};

const Header = styled('div')({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
  alignItems: 'center',
  margin: '16px',
})


const Layout = (props: Props) => {
  return (
    <Header>
      {props.children}
    </Header>
  );
};

export default Layout;
