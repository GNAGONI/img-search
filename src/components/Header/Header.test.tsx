import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Header from './index';

describe('Header', () => {
  it('should have snapshot', () => {
    const tree = renderer
      .create(
        <Header>
          <></>
        </Header>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave correctly', () => {
    const { container } = render(
      <Header>
        <button>Test</button>
      </Header>
    );

    const header = container.querySelector('div');
    expect(header).toBeInTheDocument();
    expect(header).toBeVisible();

    const button = screen.getByRole('button', { name: 'Test' });
    expect(button).toBeInTheDocument();
    expect(button).toBeVisible();
  });
});

