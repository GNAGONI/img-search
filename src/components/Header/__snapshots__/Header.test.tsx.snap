// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have snapshot 1`] = `
<div
  className="css-9xbidx"
/>
`;
